libs/ledgerjs/packages/keyring-eth/src/KeyringEth.ts

```diff
@@ -0,0 +1,40 @@
+import { Transaction } from "ethers";
+import { Keyring } from "./Keyring";
+import { LoaderOptions } from "@ledgerhq/context-module/src/shared/model/LoaderOptions";
+import { EIP712Message } from "@ledgerhq/types-live";
+
+export type EcdsaSignature = { r: `0x${string}`; s: `0x${string}`; v: number };
+
+export type EIP712Params = { domainSeparator: `0x${string}`; hashStruct: `0x${string}` };
+
+export type SignTransactionOptions = LoaderOptions["options"];
+
+export type SignMessagePayload = string | EIP712Message | EIP712Params;
```

Comment:
[ASK] Here the ticket is a bit blurry and says:
```
message: string | object, // EIP-712/EIP-191
```
I don't know what's correct so I'm just pointing it out in case you missed something.

Reply:
an EIP-712 is also an EIP-191 and here we just want to sign an EIP-712 as EIP-191 is not really used in LL.
Also, we have 2 different ways to CS an EIP-712.